src/app/components/tables/simple/terra-simple-table.component.ts

```diff
@@ -190,12 +197,14 @@ export class TerraSimpleTableComponent<D> implements OnChanges
 
     private updateHeaderCheckboxState():void
     {
-        if(this.selectedRowList.length === 0) // anything selected?
+        let selectedRows:Array<TerraSimpleTableRowInterface<D>> = this.getSelectedRows();
+
+        if(selectedRows.length === 0) // anything selected?
         {
             this.uncheckHeaderCheckbox();
         }
-        else if(this.selectedRowList.length > 0 && this.inputRowList.filter(
-                (r:TerraSimpleTableRowInterface<D>):boolean => !r.disabled).length === this.selectedRowList.length) // all selected?
+        else if(selectedRows.length > 0 && this.inputRowList.filter(
```

Comment:
Ich bin immer noch dafür die `else if`-Bedingung in eine Function auszulagern. Es erhöht definitiv die Lesbarkeit 😃